R/germline.R

```diff
@@ -93,19 +76,58 @@ take_first_allele <- function(string) {
   unlist(strsplit(string, ","))[1]
 }
 
-generate_germline_sequence <- function(v_seq, j_seq) {
-  if (is.na(v_seq) || is.na(j_seq)) {
+generate_germline_sequence <- function(seq, v_ref, v_end, cdr3_start, cdr3_end, j_start) {
+  if (is.na(seq) || is.na(v_ref) ||
+    is.na(v_end) || is.na(cdr3_start) || is.na(cdr3_end) || is.na(j_start)) {
     return(NA)
   } else {
-    return(paste(v_seq, j_seq, sep = "..."))
+    cdr3_start %<>% as.numeric()
+    cdr3_end %<>% as.numeric()
+
+    if (v_end <= cdr3_start) {
+      v_part <- v_ref
+    } else {
+      # trim intersection of V and CDR3 from reference V gene
+      v_part <- stringr::str_sub(
+        v_ref, 1,
+        max(0, stringr::str_length(v_ref) - (v_end - cdr3_start))
+      )
+    }
+
+    cdr3_part <- paste(rep("n", cdr3_end - cdr3_start), collapse = "")
```

Comment:
Wouldn't `paste0(rep("n", cdr3_end - cdr3_start))` do what's needed here?

Reply:
No, it will return a vector of 'n' instead of string

Reply:
Ok